ocaml/typing/typecore.ml

```diff
@@ -3588,6 +3588,7 @@ let rec approx_type env sty =
   match sty.ptyp_desc with
   | Ptyp_arrow (p, ({ ptyp_desc = Ptyp_poly _ } as arg_sty), sty) ->
       (* CR layouts v5: value requirement here to be relaxed *)
+      let p = Typetexp.transl_label p in
```

Comment:
This looks wrong. I think this will go awry with

```
let rec f ~(src_pos:[%src_pos]) = f ~src_pos:{ pos_fnum = ...; ... }
```

when you need type-based disambiguation to find the right `pos_fnum`. That is, `approx_type` is used to propagate typing information within a recursive definition. But by not checking for `[%src_pos]` here, you're missing an opportunity.